File: matrix-sdk-android/src/main/java/org/commonmark/ext/maths/InlineMaths.java
/*
 * Copyright (c) 2020 New Vector Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.commonmark.ext.maths;

import org.commonmark.node.CustomNode;
import org.commonmark.node.Delimited;

public class InlineMaths extends CustomNode implements Delimited {
    public enum InlineDelimiter {
        SINGLE_DOLLAR,
        ROUND_BRACKET_ESCAPED
    };

    private InlineDelimiter delimiter;

    public InlineMaths(InlineDelimiter delimiter) {
        this.delimiter = delimiter;
    }

    @Override
    public String getOpeningDelimiter() {
        switch (delimiter) {
            case SINGLE_DOLLAR:
                return "$";
            case ROUND_BRACKET_ESCAPED:
                return "\\(";
        }
        return null;
    }

    @Override
    public String getClosingDelimiter() {
        switch (delimiter) {
            case SINGLE_DOLLAR:
                return "$";
            case ROUND_BRACKET_ESCAPED:
                return "\\)";
        }
        return null;
    }
}

File: matrix-sdk-android/src/main/java/org/commonmark/ext/maths/InlineMaths.kt
/*
 * Copyright 2025 New Vector Ltd.
 * Copyright 2021 The Matrix.org Foundation C.I.C.
 *
 * SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
 * Please see LICENSE files in the repository root for full details.
 */
package org.commonmark.ext.maths

import org.commonmark.node.CustomNode
import org.commonmark.node.Delimited

internal class InlineMaths(private val delimiter: InlineDelimiter) : CustomNode(), Delimited {
    enum class InlineDelimiter {
        SINGLE_DOLLAR,
        ROUND_BRACKET_ESCAPED
    }

    override fun getOpeningDelimiter(): String {
        return when (delimiter) {
            InlineDelimiter.SINGLE_DOLLAR -> "$"
            InlineDelimiter.ROUND_BRACKET_ESCAPED -> "\\("
        }
    }

    override fun getClosingDelimiter(): String {
        return when (delimiter) {
            InlineDelimiter.SINGLE_DOLLAR -> "$"
            InlineDelimiter.ROUND_BRACKET_ESCAPED -> "\\)"
        }
    }
}
